Allow DSL scripts for script transformation

J-N-K opened the pull request:
This extends the DSL scriptengine to accept `inputString` as implicit variable in the engine context. By that DSL scripts can be used for the script transformation. 

Signed-off-by: Jan N. Klug <github@klug.nrw>

kaikreuzer commented:
The current [JS transformation](https://www.openhab.org/addons/transformations/javascript/) uses `input` as a variable name for the input. Shouldn't we align the naming across all scripting languages?

kaikreuzer commented:
The current [JS transformation](https://www.openhab.org/addons/transformations/javascript/) uses `input` as a variable name for the input. Shouldn't we align the naming across all scripting languages?

J-N-K commented:
Good point. I‘ll change that for the script transformation in general, since it was only merged a milestone ago or so, I think we can still accept this breaking change. I can do that this afternoon.

J-N-K commented:
@kaikreuzer I have changed it to `input` and also added a check that the file/configuration type is `script`. The script type itself (i.e. js/dsl/...) is still prepended to the filename (as suggested in the other PR).

kaikreuzer commented on bundles/org.openhab.core.automation.module.script/src/test/java/org/openhab/core/automation/module/script/ScriptTransformationServiceTest.java:
```diff
@@ -46,13 +46,18 @@
 @MockitoSettings(strictness = Strictness.LENIENT)
 public class ScriptTransformationServiceTest {
-    private static final String SCRIPT_TYPE = "script";
+    private static final String SCRIPT_LANGUAGE = "script";
```
Why is language set to "script"? That does not seem to fit...

kaikreuzer commented on bundles/org.openhab.core.automation.module.script/src/test/java/org/openhab/core/automation/module/script/ScriptTransformationServiceTest.java:
```diff
@@ -65,56 +70,64 @@ public class ScriptTransformationServiceTest {
+        when(scriptEngineManager.createScriptEngine(eq(SCRIPT_LANGUAGE), any())).thenReturn(scriptEngineContainer);
         when(scriptEngineManager.isSupported(anyString()))
-                .thenAnswer(scriptType -> SCRIPT_TYPE.equals(scriptType.getArgument(0)));
+                .thenAnswer(scriptType -> SCRIPT_LANGUAGE.equals(scriptType.getArgument(0)));
```
Script language is here compared to script type now - could the naming be aligned?

J-N-K commented:
Done.

J-N-K commented:
@kaikreuzer Can we merge here? This should be part of 3.3 so there are less breaking changes after the release.

kaikreuzer reviewed (approved):
Sure, thank you!
